llvm/test/CodeGen/X86/atomic-non-integer.ll

```diff
@@ -787,3 +787,100 @@ define double @load_double_seq_cst(ptr %fptr) {
   %v = load atomic double, ptr %fptr seq_cst, align 8
   ret double %v
 }
+
+define void @store_bfloat(ptr %fptr, bfloat %v) {
+; X86-LABEL: store_bfloat:
+; X86:       # %bb.0:
+; X86-NEXT:    movl {{[0-9]+}}(%esp), %eax
+; X86-NEXT:    movzwl {{[0-9]+}}(%esp), %ecx
+; X86-NEXT:    movw %cx, (%eax)
+; X86-NEXT:    retl
+;
+; X64-SSE-LABEL: store_bfloat:
+; X64-SSE:       # %bb.0:
+; X64-SSE-NEXT:    pextrw $0, %xmm0, %eax
+; X64-SSE-NEXT:    movw %ax, (%rdi)
+; X64-SSE-NEXT:    retq
+;
+; X64-AVX-LABEL: store_bfloat:
+; X64-AVX:       # %bb.0:
+; X64-AVX-NEXT:    vpextrw $0, %xmm0, %eax
+; X64-AVX-NEXT:    movw %ax, (%rdi)
+; X64-AVX-NEXT:    retq
+  store atomic bfloat %v, ptr %fptr unordered, align 2
+  ret void
+}
+
+; Work around issue #92899 by casting to float
```

Comment:
Maybe you can put it in a separate file and remove SSE1 test.

Reply:
I'm not putting in that effort; Instead of fracturing tests it's better to just fix the backend

Reply:
It's fixable, and it's not even hard to fix. You can always just bitcast to i16, just like half is already handled. 

An "external function" compiled the same way in the IR will work. The lack of defined C ABI is not relevant on getting the IR into a working state. You can always implement something

Reply:
That cannot be called a fix. It's not a simple "external function" problem, but the basestone functions like `__truncsfhf2`, `__extendhfsf2` that used by almost all operations. They are provided in library with standard ABI. You cannot make a meaningful application work without them.

The half has its own story. It's being supported in i16 prior to ABI definition. We leave it unchanged for ABI undefined case doesn't mean we have to duplicate it for new type.

Reply:
#93146 fixes this. Whether or not compiler-rt is wired up correctly to use the same IR type is a separate problem, but the IR has a clear handleable meaning